tutor/src/screens/assignment-review/ux.js

```diff
@@ -312,9 +315,11 @@ export default class AssignmentReviewUX {
 
     @action.bound async onSavePlan() {
         await this.editUX.savePlan();
-        Object.assign(this.planScores, pick(this.editUX.plan, ['title', 'description']));
-        this.planScores.grading_template = this.editUX.plan.gradingTemplate;
-        this.isDisplayingEditAssignment = false;
+        runInAction(() => {
```

Comment:
I didn't realize we'd need to use `runInAction` inside something decorated with `@action`? Or is this to process everything a single transaction?

Reply:
yah this is a tricky one, the function is async, which means babel will actually splits it into multiple functions, and the code after the `await` will not be enclosed inside the main action.   https://mobx.js.org/actions.html#asynchronous-actions explains it a bit more

Reply:
there's a new way to do this using `flow` but I haven't explored it yet